Broken upgrade from RC14 to RC15

## Expected Behavior
Working upgrade from RC14 to RC15

## Observed Behavior
<!---Tell us what happens instead of the expected behavior -->

```
$ sh gradlew -S
> Task :detekt FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt'.
> A problem occurred starting process 'command 'C:\Program Files\Java\jdk1.8.0_121\bin\java.exe''

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':detekt'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:151)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:148)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:191)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:141)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionStateTaskExecuter.execute(ResolveBeforeExecutionStateTaskExecuter.java:75)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:108)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionOutputsTaskExecuter.execute(ResolveBeforeExecutionOutputsTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.StartSnapshotTaskInputsBuildOperationTaskExecuter.execute(StartSnapshotTaskInputsBuildOperationTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveAfterPreviousExecutionStateTaskExecuter.execute(ResolveAfterPreviousExecutionStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:94)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:49)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:49)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'C:\Program Files\Java\jdk1.8.0_121\bin\java.exe''
        at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:232)
        at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:209)
        at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:356)
        at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:86)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        ... 6 more
Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'C:\Program Files\Java\jdk1.8.0_121\bin\java.exe'
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
        at net.rubygrapefruit.platform.internal.WindowsProcessLauncher.start(WindowsProcessLauncher.java:22)
        at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
        at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:97)
        at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:70)
        ... 7 more
Caused by: java.io.IOException: Cannot run program "C:\Program Files\Java\jdk1.8.0_121\bin\java.exe" (in directory "C:\Users\krystian.panek\Projects\gradle-aem-plugin"): CreateProcess error=206, The filename or extension is too long
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
        ... 11 more
Caused by: java.io.IOException: CreateProcess error=206, The filename or extension is too long
        at java.lang.ProcessImpl.create(Native Method)
        at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
        at java.lang.ProcessImpl.start(ProcessImpl.java:137)
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
        ... 12 more


* Get more help at https://help.gradle.org

BUILD FAILED in 5s
7 actionable tasks: 1 executed, 1 from cache, 5 up-to-date
```

switching back to RC14 resolves problem

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

n/a

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

previously working with RC14, tried to upgrade to RC15 without changing detekt.yml config

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used: RC15
* Version of Gradle used (if applicable): 5.4.0
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): https://github.com/Cognifide/gradle-aem-plugin/blob/master/buildSrc/build.gradle.kts#L11

I just got similar on OSX: 
```

Caused by: java.io.IOException: Cannot run program "/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home/bin/java" (in directory "/Users/eugenmartynov/Development/source/ing/lovebird-android"): error=7, Argument list too long
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
        ... 7 more
Caused by: java.io.IOException: error=7, Argument list too long
        ... 8 more

```

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
@@ -230,7 +230,8 @@ open class Detekt : SourceTask(), VerificationTask {
             project = project,
             arguments = arguments.toList(),
             ignoreFailures = ignoreFailuresProp.getOrElse(false),
-            classpath = detektClasspath.plus(pluginClasspath)
+            classpath = detektClasspath.plus(pluginClasspath),
+            taskName = name
         )
 
         if (xmlReportTargetFileOrNull != null) {

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
@@ -229,7 +229,6 @@ open class Detekt : SourceTask(), VerificationTask {
         DetektInvoker.invokeCli(
             project = project,
             arguments = arguments.toList(),
-            debug = debugOrDefault,
             ignoreFailures = ignoreFailuresProp.getOrElse(false),
             classpath = detektClasspath.plus(pluginClasspath)
         )

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
@@ -130,7 +130,8 @@ open class DetektCreateBaselineTask : SourceTask() {
             project = project,
             arguments = arguments.toList(),
             ignoreFailures = ignoreFailures.getOrElse(false),
-            classpath = detektClasspath.plus(pluginClasspath)
+            classpath = detektClasspath.plus(pluginClasspath),
+            taskName = name
         )
     }
 }

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
@@ -129,7 +129,6 @@ open class DetektCreateBaselineTask : SourceTask() {
         DetektInvoker.invokeCli(
             project = project,
             arguments = arguments.toList(),
-            debug = debug.getOrElse(false),
             ignoreFailures = ignoreFailures.getOrElse(false),
             classpath = detektClasspath.plus(pluginClasspath)
         )

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt
@@ -37,6 +37,6 @@ open class DetektGenerateConfigTask : SourceTask() {
             InputArgument(source)
         )
 
-        DetektInvoker.invokeCli(project, arguments.toList(), detektClasspath)
+        DetektInvoker.invokeCli(project, arguments.toList(), detektClasspath, name)
     }
 }

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -16,14 +16,19 @@ object DetektInvoker {
         debug: Boolean = false,
         ignoreFailures: Boolean = false
     ) {
+        val detektTmpDir = project.mkdir("${project.buildDir}/tmp/detekt")
+        val argsFile = project.file("$detektTmpDir/args")
+
         val cliArguments = arguments.flatMap(CliArgument::toArgument)
+
+        argsFile.writeText(cliArguments.joinToString("\n"))
 
         if (debug) println(cliArguments)
 
         val proc = project.javaexec {
             it.main = DETEKT_MAIN
             it.classpath = classpath
-            it.args = cliArguments
+            it.args = listOf("@${argsFile.absolutePath}")
             it.isIgnoreExitValue = true
         }
         val exitValue = proc.exitValue

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -13,10 +13,11 @@ object DetektInvoker {
         project: Project,
         arguments: List<CliArgument>,
         classpath: FileCollection,
+        taskName: String,
         ignoreFailures: Boolean = false
     ) {
         val detektTmpDir = project.mkdir("${project.buildDir}/tmp/detekt")
-        val argsFile = project.file("$detektTmpDir/args")
+        val argsFile = project.file("$detektTmpDir/$taskName.args")
 
         val cliArguments = arguments.flatMap(CliArgument::toArgument)
 

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -13,7 +13,6 @@ object DetektInvoker {
         project: Project,
         arguments: List<CliArgument>,
         classpath: FileCollection,
-        debug: Boolean = false,
         ignoreFailures: Boolean = false
     ) {
         val detektTmpDir = project.mkdir("${project.buildDir}/tmp/detekt")
@@ -23,7 +22,7 @@ object DetektInvoker {
 
         argsFile.writeText(cliArguments.joinToString("\n"))
 
-        if (debug) println(cliArguments)
+        project.logger.debug(cliArguments.joinToString(" "))
 
         val proc = project.javaexec {
             it.main = DETEKT_MAIN
@@ -32,7 +31,7 @@ object DetektInvoker {
             it.isIgnoreExitValue = true
         }
         val exitValue = proc.exitValue
-        if (debug) println("Detekt finished with exit value $exitValue")
+        project.logger.debug("Detekt finished with exit value $exitValue")
 
         when (exitValue) {
             1 -> throw GradleException("There was a problem running detekt.")

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/invoke/DetektInvoker.kt
@@ -1,7 +1,5 @@
 package io.gitlab.arturbosch.detekt.invoke
 
-import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT
-import io.gitlab.arturbosch.detekt.CONFIGURATION_DETEKT_PLUGINS
 import org.gradle.api.GradleException
 import org.gradle.api.Project
 import org.gradle.api.file.FileCollection

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskMultiModuleConsolidationTest.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskMultiModuleConsolidationTest.kt
@@ -67,8 +67,6 @@ internal class DetektTaskMultiModuleConsolidationTest : Spek({
         it("using the kotlin dsl") {
 
             val mainBuildFileContent: String = """
-                |import io.gitlab.arturbosch.detekt.detekt
-                |
                 |plugins {
                 |   `java-library`
                 |    id("io.gitlab.arturbosch.detekt")

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskMultiModuleTest.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskMultiModuleTest.kt
@@ -67,8 +67,6 @@ internal class DetektTaskMultiModuleTest : Spek({
             it("can be done using the kotlin dsl") {
 
                 val mainBuildFileContent: String = """
-				|import io.gitlab.arturbosch.detekt.detekt
-				|
 				|plugins {
 				|   `java-library`
 				|	id("io.gitlab.arturbosch.detekt")
@@ -138,8 +136,6 @@ internal class DetektTaskMultiModuleTest : Spek({
             it("can be done using the kotlin dsl") {
 
                 val mainBuildFileContent: String = """
-				|import io.gitlab.arturbosch.detekt.detekt
-				|
 				|plugins {
 				|   `java-library`
 				|	id("io.gitlab.arturbosch.detekt")
@@ -209,8 +205,6 @@ internal class DetektTaskMultiModuleTest : Spek({
             it("can be done using the kotlin dsl") {
 
                 val mainBuildFileContent: String = """
-				|import io.gitlab.arturbosch.detekt.detekt
-				|
 				|plugins {
 				|   `java-library`
 				|	id("io.gitlab.arturbosch.detekt")
@@ -288,8 +282,6 @@ internal class DetektTaskMultiModuleTest : Spek({
             it("can be done using the kotlin dsl") {
 
                 val mainBuildFileContent: String = """
-				|import io.gitlab.arturbosch.detekt.detekt
-				|
 				|plugins {
 				|   `java-library`
 				|	id("io.gitlab.arturbosch.detekt")

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DslGradleRunner.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DslGradleRunner.kt
@@ -90,7 +90,7 @@ class DslGradleRunner(
     }
 
     private fun buildGradleRunner(tasks: List<String>): GradleRunner {
-        val args = listOf("--stacktrace", "--info", "--build-cache") + tasks.toList()
+        val args = listOf("--stacktrace", "--info", "--build-cache", "--debug") + tasks.toList()
 
         return GradleRunner.create().apply {
             withProjectDir(rootDir)

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DslTestBuilder.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DslTestBuilder.kt
@@ -76,8 +76,6 @@ abstract class DslTestBuilder {
     private class KotlinBuilder : DslTestBuilder() {
         override val gradleBuildName: String = "build.gradle.kts"
         override val gradleBuildConfig: String = """
-                |import io.gitlab.arturbosch.detekt.detekt
-                |
                 |plugins {
                 |   `java-library`
                 |    id("io.gitlab.arturbosch.detekt")

